src/component/vocabulary/RemoveVocabularyDialog.tsx

```diff
@@ -0,0 +1,239 @@
+        </li>
+      );
+    });
+
+  const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setconfirmationVocabularyName(e.currentTarget.value);
+  };
+
+  const isConfirmDisabled =
+    !canRemoveSilently &&
+    (!canRemove ||
+      confirmationVocabularyName !== props.vocabulary.getLabel(locale));
+
+  const validateInput = () => {
+    return isConfirmDisabled
+      ? ValidationResult.BLOCKER
+      : ValidationResult.VALID;
+  };
+
+  const renderRemoveWarning = () => (
+    <>
+      <Label>
+        {formatMessage(
+          "vocabulary.remove.dialog.text." +
+            (canRemoveSilently ? "empty" : "nonEmpty"),
+          {
+            b,
+          }
+        )}
+        <br />
+        <b>{i18n("vocabulary.remove.dialog.text.permanent")}</b>
+      </Label>
+
+      <If expression={!canRemoveSilently}>
+        <p>
+          {formatMessage("vocabulary.remove.dialog.text.termCount", {
+            b,
+            count: props.vocabulary.termCount,
+          })}
+        </p>
+        <CustomInput
+          label={i18n("vocabulary.remove.dialog.input.label")}
+          value={confirmationVocabularyName}
+          onChange={onInputChange}
+          validation={validateInput()}
+        />
+      </If>
+    </>
+  );
+
+  const renderUnableToRemove = () => {
+    return (
+      <>
+        <Label>
+          {i18n("vocabulary.remove.dialog.relations.error.cantRemove")}
+        </Label>
+        <p>
+          <If expression={vocabularyRelations.length > 0}>
+            {formatMessage(
+              "vocabulary.remove.dialog.relations.error.vocabularyRelations",
+              { vocabularyRelations: vocabularyRelations.length, b }
+            )}
+            <br />
+          </If>
+          <If expression={termsRelations.length > 0}>
+            {formatMessage(
+              "vocabulary.remove.dialog.relations.error.termsRelations",
+              { termsRelations: termsRelations.length, b }
+            )}
+            <br />
+          </If>
+        </p>
+        <h3>{props.vocabulary.getLabel(locale)}</h3>
+        <details>
+          <summary>{i18n("vocabulary.remove.dialog.relations")}</summary>
+          <ul>
+            {renderStatementArray(
+              vocabularyRelations,
+              "remove-vocabulary-dialog-vocabulary-relation-",
+              VocabularyIriLink
+            )}
+          </ul>
+          <ul>
+            {renderStatementArray(
+              termsRelations,
+              "remove-vocabulary-dialog-term-relation-",
+              TermIriLink
+            )}
+          </ul>
+        </details>
+      </>
+    );
+  };
+
+  const renderContent = () => {
+    if (canRemove) {
+      return renderRemoveWarning();
+    }
+    return renderUnableToRemove();
+  };
+
+  return (
+    <ConfirmCancelDialog
+      show={props.show}
+      id="remove-vocabulary-dialog"
+      onClose={cancel}
+      onConfirm={submit}
+      confirmColor={"outline-danger"}
```

Comment:
Why is the primary confirm button `outline-danger`? It is inconsistent with other remove dialogs in TermIt

Reply:
So wouldn't it be better to change all remove dialogs to use the danger color for the remove button? (@MichalMed )
![image](https://github.com/user-attachments/assets/37e9ed55-1d30-41c5-8a49-b9937489b5aa)

![image](https://github.com/user-attachments/assets/f22197c5-79b5-4944-bb24-f33b2838c30e)

Reply:
In that case, I would go with outline-danger everywhere to make them the same as the remove action button.